packages/audio-block/src/AudioBlock.tsx

```diff
@@ -0,0 +1,210 @@
+    ]);
+
+    const saveTitle = (value: string) => {
+        if (value !== blockSettings.title) {
+            setBlockSettings({
+                title: value,
+            });
+        }
+    };
+
+    const saveDescription = (value: string) => {
+        if (value !== blockSettings.description) {
+            setBlockSettings({
+                description: value,
+            });
+        }
+    };
+
+    const onRemoveAsset = () => {
+        deleteAssetIdsFromKey(AUDIO_ID, [audio?.id]);
+    };
+
+    const updateAudioAsset = (audio: Asset) => {
+        updateAssetIdsFromKey(AUDIO_ID, [audio.id]).then(() => setIsLoading(false));
+    };
+
+    const openAssetChooser = () => {
+        appBridge.openAssetChooser(
+            async (result) => {
+                setIsLoading(true);
+                updateAudioAsset(result[0]);
+                appBridge.closeAssetChooser();
+            },
+            {
+                multiSelection: false,
+                objectTypes: [AssetChooserObjectType.File],
+                extensions: FileExtensionSets.Audio,
+            }
+        );
+    };
+
+    useEffect(() => {
+        if (selectedFiles) {
+            setIsLoading(true);
+            uploadFile(selectedFiles[0]);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedFiles]);
+
+    useEffect(() => {
+        if (droppedFiles) {
+            if (FileExtensionSets.Audio.includes(droppedFiles[0].name.split('.').pop() as FileExtension)) {
+                setIsLoading(true);
+                uploadFile(droppedFiles[0]);
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [droppedFiles]);
+
+    useEffect(() => {
+        if (doneAll && uploadResults) {
+            updateAudioAsset(uploadResults[0]);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [doneAll, uploadResults]);
+
+    return (
+        <div data-test-id="audio-block" className={audioBlockClassNames}>
+            {audio ? (
+                <div className="tw-group">
+                    {isEditing && (
+                        <ItemToolbar
+                            onRemoveAsset={onRemoveAsset}
+                            onUploadClick={openFileDialog}
+                            onAssetChooseClick={openAssetChooser}
+                        />
+                    )}
+
+                    {isLoading ? (
+                        <div className="tw-flex tw-items-center tw-justify-center tw-h-14">
+                            <LoadingCircle />
+                        </div>
+                    ) : (
+                        <audio
+                            data-test-id="audio-block-audio-tag"
+                            key={audio.id}
+                            controls
+                            className={audioTagClassNames}
+                            controlsList="nodownload"
+                            preload="auto"
```

Comment:
You are completely right. I changed it to `preload="metadata"`. In my opinion its even better like that because it just loads audio file if you really play it, otherwise there are just the metadata loaded, [commit](https://github.com/Frontify/guideline-blocks/pull/475/commits/22d4391973d6e2eaac4785fca017b089ddbfe2a1)